feat(docs): Improve the toolchain setup page (#2272)
zmkfirmware/zmk: docs/docs/development/setup/getting-started.md

---
title: Getting Started
sidebar_label: Getting Started
---

:::info
Throughout this documentation you will see commands like

```sh
sudo apt update
```

These commands should be run in a terminal such as Bash (Linux/MacOS/Docker), PowerShell, or Command Prompt (Windows).

> :::

Reviewer: Do you think this is necessary? I feel like this level of detail doesn't belong in dev documentation. I'd probably not recommend the user to use the local build option if this is a point of confusion.

Author: It depends on the target audience. Anyone with dev experience shouldn't need this note, but I think it may be helpful for some less experienced windows users. It does contribute to making it too verbose though, so I'll get rid of it.

:::tip
We recommend reading through the setup process before following it step by step, to ensure that you are happy with installing the required dependencies.
:::

## Environment Setup

> There are two ways to setup the ZMK development environment:
> 
> - [Docker](/docs/development/setup/docker) ;
> - [Native](/docs/development/setup/native), i.e. using the operating system directly.

Reviewer: ```suggestion
There are two ways to setup the ZMK development environment: using [Docker](docker.md) or using a [native setup](native.mdx).
```
I'd say we can keep this small since the details are already elaborated on the following paragraphs.

(We also prefer file path links in the rest of the docs)

Author: Will change to file path links, but I think the bullet points break up the text and make it easier to read. Otherwise it's just a block of text in this section. Really, I'd prefer to break up the following paragraphs a bit to be more visually appealing, but can't think of a good way to do so.

The Docker approach is a self-contained development environment while the native approach will setup your local operating system for development. It uses the same [Docker image which is used by the GitHub action](https://github.com/zmkfirmware/zmk-docker) for local development. Beyond the benefits of [dev/prod parity](https://12factor.net/dev-prod-parity), this approach may be easier to set up for some operating systems. No toolchain or dependencies are necessary when using Docker; the container image has the toolchain installed and set up to use.

The native approach installs the toolchain and dependencies on your system directly. This typically runs slightly faster than the Docker approach, and can be preferable for users who already have the dependencies on their system.

> Please see the [Docker](/docs/development/setup/docker) instructions or [native](/docs/development/setup/native) instructions to continue setup.

Reviewer: ```suggestion
Please see the [Docker](docker.md) instructions or [native](native.mdx) instructions to continue setup.
```
